spec/features/admin/index_spec.rb

```diff
@@ -18,27 +18,27 @@
-    invoice4 = Invoice.create!(status: "completed", customer_id: customer4.id)
-    invoice5 = Invoice.create!(status: "completed", customer_id: customer5.id)
-    invoice6 = Invoice.create!(status: "completed", customer_id: customer6.id)
-    invoice7 = Invoice.create!(status: "completed", customer_id: customer6.id)
-    invoice_item1 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice1.id, quantity: 5, unit_price: 13635, status: "pending")
-    invoice_item2 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice2.id, quantity: 9, unit_price: 23324, status: "pending")
-    invoice_item3 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice3.id, quantity: 9, unit_price: 23324, status: "pending")
-    invoice_item4 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice4.id, quantity: 9, unit_price: 23324, status: "pending")
-    invoice_item5 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice5.id, quantity: 9, unit_price: 23324, status: "pending")
-    invoice_item6 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice6.id, quantity: 9, unit_price: 23324, status: "pending")
-    invoice_item7 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice6.id, quantity: 9, unit_price: 23324, status: "pending")
-    transaction1 = Transaction.create!(invoice_id: invoice1.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "failed")
-    transaction2 = Transaction.create!(invoice_id: invoice2.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
-    transaction3 = Transaction.create!(invoice_id: invoice3.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
-    transaction4 = Transaction.create!(invoice_id: invoice4.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
-    transaction5 = Transaction.create!(invoice_id: invoice5.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
-    transaction6 = Transaction.create!(invoice_id: invoice6.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
-    transaction7 = Transaction.create!(invoice_id: invoice7.id, credit_card_number: "1234567890987654", credit_card_expiration_date: "04/27", result: "success")
+    @invoice1 = Invoice.create!(status: "in progress", customer_id: customer1.id)
+    @invoice2 = Invoice.create!(status: "in progress", customer_id: customer2.id)
+    @invoice3 = Invoice.create!(status: "in progress", customer_id: customer3.id)
+    @invoice4 = Invoice.create!(status: "in progress", customer_id: customer4.id)
+    @invoice5 = Invoice.create!(status: "completed", customer_id: customer5.id)
+    @invoice6 = Invoice.create!(status: "completed", customer_id: customer6.id)
+    @invoice7 = Invoice.create!(status: "completed", customer_id: customer6.id)
```

Comment:
With the invoices and invoice_items :status columns changed to integer, should status: "In progress" be changed to and integer?

Reply:
As far as I know, our csv's have strings for statuses, when we rake them into our database, which holds statuses as integers, the enum line in the model converts each string into its integer value given the indexed array. So if a item in our csv has a status like "pending" pending will get matched to a number like 0,1,2, or 3, then when we request information back out with activerecord or w/e its converted back into string name. So the database only knows about status integers and we only really know about status strings and our model is the "man in the middle" that converts them all. We can play with the data and set the status of some invoice to 0, 1, 2, or 3 and when we query the item again, it will still come back as a string but it will be the string associated with whatever number you changed it too.